### ISSUE TITLE
Multiyear Scheduling

### ISSUE THREAD

**Issue by brianlball** [NREL/EnergyPlus#6898]
Need ability to run multiyear (including mid year start and end dates) run periods with schedulefiles.  Use cases are calibration runs with metered data that spans multi/mid years.  Currently schedulefiles are limited to 8760 and must start at jan 1.  would be great if these constraints could be relaxed.

**Comment by asparke2** [comment]
Agree.  Another use case I have run across is model predictive control.

**Comment by jasondegraw** [comment]
There's a related issue in #4154, there's a general lack of capability in doing multi-year scheduling. There are probably EMS solutions, but it makes more sense to me to add the ability to input multi-year schedules in schedule files and in the IDF. @JasonGlazer, do you have any thoughts?

**Comment by JasonGlazer** [comment]
I think we need to be careful with how the year is included. Most users of EnergyPlus in most use cases, utilize typical weather files which do not have specific years associated with them. It is already a somewhat confusing topic since most of the typical weather files are made up of months from different actual weather years. We should try not to make it any more confusing when a year is mentioned. It should not be a required input and perhaps the objects that include years should be completely separate so that current users are not confused. This is probably a bigger deal for normal schedule IDF objects. Maybe make new object called something like Schedule:Year:Multiple and Schedule:Compact:MultipleYear. On the other hand, I hate to see a proliferation of objects that do the same function. I'm not sure.

For Schedule:File, I would probably just add new fields that are optional for the year. People who use that object are already more advanced users.  

When you do include the year, make sure a check is made for the Day of Week for Start Day in the RunPeriod object since that should either be blank when a real year is used or should be consistent.

Another question is whether all schedules need to be specify the year, if any of them do. 

@mjwitte would probably like to weigh in on this topic also.

**Comment by mjwitte** [comment]
Actual years and start day validation will be introduced with #6478 [modifications to RunPeriod](https://github.com/NREL/EnergyPlus/pull/6478/files#diff-38f7e3f8a464288a615843b843ce0ff1R25), so that can be assumed here. The input changes should be relatively minor for this. The bigger lift will be in the schedule manager to align data and dates. Ideally, Schedule:File should support an optional date/time column(s) that can automatically align the schedule data with the simulation.

**Comment by Lefort-Antoine** [comment]
It appears that #6478 have been solved and include in energyplus release 9.0.1. [https://github.com/NREL/EnergyPlus/releases/tag/v9.0.1](url)
However the constraints have not been relaxed in IDD for schedule:file and it not availaable to specify year into schedule:compact model.
How can we specify multi year schedule with different value depending on year ?

**Comment by mjwitte** [comment]
Until this feature is added, one approach would be to have a separate Schedule:File (or Schedule:Compact) object for each year and then use EMS to grab from the correct year to override a base schedule. If using Schedule:File, each year could just be a different column in the same file. `Year` is already available as a built-in EMS variable, so this should be possible.
